docu/Concepts/TechnicalRequirements/UC_Introduction_of_Gradido-ID.md

```diff
@@ -0,0 +1,141 @@
+# Introduction of Gradido-ID
+
+## Motivation
+
+To introduce the Gradido-ID base on the requirement to identify an user account per technical key instead of using an email-address. Such a technical key ensures an exact identification of an user account without giving detailed information for possible missusage.
+
+Additionally the Gradido-ID allows to administrade any user account data like changing the email address or define several email addresses without any side effects on the identification of the user account.
+
+## Definition
+
+The formalized definition of the Gradido-ID can be found in the document [BenutzerVerwaltung#Gradido-ID](../BusinessRequirements/BenutzerVerwaltung#Gradido-ID).
+
+## Steps of Introduction
+
+To Introduce the Gradido-ID there are several steps necessary. The first step is to define a proper database schema with additional columns and tables followed by data migration steps to add or initialize the new columns and tables by keeping valid data at all.
+
+The second step is to decribe all concerning business logic processes, which have to be adapted by introducing the Gradido-ID.
+
+### Database-Schema
+
+#### Users-Table
+
+The entity users has to be changed by adding the following columns.
+
+| Column                   | Type   | Description                                                                            |
+| ------------------------ | ------ | -------------------------------------------------------------------------------------- |
+| gradidoID                | String | technical unique key of the user as UUID (version 4)                                   |
+| alias                    | String | a business unique key of the user                                                      |
+| passphraseEncryptionType | int    | defines the type of encrypting the passphrase: 1 = email (default), 2 = gradidoID, ... |
+| emailID                  | int    | technical foreign key to the new entity Contact                                        |
+
+##### Email vs emailID
+
+The existing column `email`, will now be changed to the primary email contact, which will be stored as a contact entry in the new `UserContacts` table. It is necessary to decide if the content of the `email `will be changed to the foreign key `emailID `to the contact entry with the email address or if the email itself will be kept as a denormalized and duplicate value in the `users `table.
```

Comment:
this is a completely different feature.... multi-email has nothing todo with GradidoID. Even the user-alias is a far fetch from the original statement "GradidoID"

Reply:
That's correct, introducing the GradidoID could be a simply new added attribute.
But...
Introducing a new primary identifier for an user, has effect on the existing primary identifier 'email'.
And what will happen with the existing identifier, if it will be not the primary, but the second or third ranged identifier?
On my opinion the concept for the new GradidoID have to describe the possible changes for other identifiers like email and alias too.

Let us devide this concept perhaps in separated tickets, which can be implemented sequencially for different releases. 
But for the migration efforts it will be more efficient to do all migration at once. For sure the risk of errors will rise up the more changes are done during migration.

Reply:
I am a fan of splitting up things ;) KISS